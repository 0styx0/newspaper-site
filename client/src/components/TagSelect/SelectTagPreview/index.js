import React from 'react';
import { Redirect } from 'react-router'

import TagSelect from '../';


export default class SelectTagPreview extends React.Component {

    constructor() {
        super();

        this.state = {
            redirect: ""
        }
        this.onChange = this.onChange.bind(this);
    }

    onChange(event) {


        this.setState({
            redirect: `/tag/${event.target.value}`
        })
    }


    render() {

        return  <span key={this.state.redirect}>
                    <TagSelect
                        tags={["Current Issue"]}
                        props={{
                            defaultValue: window.location.pathname.split("/")[2] || "../",
                            onChange: this.onChange
                        }}
                    />
                    {this.state.redirect ? <Redirect to={this.state.redirect} /> : ""}
                </span>

    }
}

