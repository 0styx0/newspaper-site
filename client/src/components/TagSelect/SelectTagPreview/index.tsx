import * as React from 'react';
import { Redirect } from 'react-router'

import TagSelect from '../';

interface State {
    redirect: string;
};

export default class SelectTagPreview extends React.Component<{}, State> {

    constructor() {
        super();

        this.state = {
            redirect: ""
        }
        this.onChange = this.onChange.bind(this);
    }

    onChange(event: Event) {


        this.setState({
            redirect: `/tag/${(event.target as HTMLSelectElement).value}`
        })
    }


    render() {

        return  <span key={this.state.redirect}>
                    <TagSelect
                        tags={["Current Issue"]}
                        props={{
                            defaultValue: window.location.pathname.split("/")[2] || "../",
                            onChange: this.onChange
                        }}
                    />
                    {this.state.redirect ? <Redirect to={this.state.redirect} /> : ""}
                </span>

    }
}

