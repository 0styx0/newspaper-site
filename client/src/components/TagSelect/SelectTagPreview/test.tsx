import * as React from 'react';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import SelectTagPreview from './';
import { mount } from 'enzyme';

describe('<Slideshow>', () => {

    function setup(images: Image[]) {

        return mount(
            <MemoryRouter>
                <SelectTagPreview />
            </MemoryRouter>
        );
    }

    describe('snapshots', () => {

        it('renders correctly', () => {

            const tree = renderer.create(
                <MemoryRouter>
                    <SelectTagPreview />
                </MemoryRouter>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('#onChange', () => {

        it('redirects to e.target.value', () => {

            const wrapper = setup();
            const component = (wrapper.find(SelectTagPreview) as any).node;

            expect(component.state.redirect).toBe('');

            const optionToChange = wrapper.find('option').first();

            optionToChange.simulate('change');

            expect(component.state.redirect).toBe(`/tag/${optionToChange.node.value}`);
        });
    });
});