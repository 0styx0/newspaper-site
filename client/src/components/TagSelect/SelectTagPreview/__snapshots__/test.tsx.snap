// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slideshow> snapshots renders correctly 1`] = `
<span>
  <select
    defaultValue="../"
    onChange={[Function]}
  >
    <option
      value="Current Issue"
    >
      Current Issue
    </option>
    <option
      value="news"
    >
      News
    </option>
    <option
      value="reaction"
    >
      Reaction
    </option>
    <option
      value="opinion"
    >
      Opinion
    </option>
    <option
      value="poll"
    >
      Poll
    </option>
    <option
      value="features"
    >
      Features
    </option>
    <option
      value="sports"
    >
      Sports
    </option>
    <option
      value="politics"
    >
      Politics
    </option>
    <option
      value="other"
    >
      Other
    </option>
  </select>
  
</span>
`;
